lupa/_lupa.pyx

```diff
@@ -1413,21 +1415,34 @@ cdef tuple unpack_multiple_lua_results(LuaRuntime runtime, lua_State *L, int nar
 
 # ref-counting support for Python objects
 
+@cython.final
+@cython.internal
+@cython.freelist(8)
+cdef class _PyReference:
+    cdef object obj
+    cdef uintptr_t refcnt
```

Comment:
I wonder if the refcount could be stored in the `py_object` struct, instead of using a Python class for it?
(I guess the `WrappedObjectFlags` might get in the way, hopefully not too much…)

Reply:
See the new commits, how we don't need to keep the reference count anymore.